lib/pages/homeserver_picker/homeserver_picker_view.dart

```diff
@@ -148,79 +152,107 @@ class HomeserverTextField extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
-    return TypeAheadField(
-      textFieldConfiguration: TextFieldConfiguration(
-        autofocus: controller.state != HomeserverState.ssoLoginServer,
-        autocorrect: false,
-        enabled: true,
-        controller: controller.homeserverController,
-        decoration: InputDecoration(
-          border: OutlineInputBorder(
-            borderSide:
-                BorderSide(color: Theme.of(context).colorScheme.outline),
-          ),
-          focusedBorder: controller.state == HomeserverState.wrongServerName
-              ? OutlineInputBorder(
-                  borderSide: BorderSide(
-                    color: Theme.of(context).colorScheme.error,
-                    width: 2.0,
-                  ),
-                )
-              : null,
-          labelText: controller.state != HomeserverState.wrongServerName
-              ? L10n.of(context)!.homeserver
-              : L10n.of(context)!.wrongServerName,
-          labelStyle: Theme.of(context).textTheme.bodySmall?.copyWith(
-                letterSpacing: 0.4,
-                color: controller.state != HomeserverState.wrongServerName
-                    ? Theme.of(context).colorScheme.onSurface
-                    : Theme.of(context).colorScheme.error,
+    final useShortCuts = (PlatformInfos.isWeb ||
+        PlatformInfos.isDesktop ||
+        AppConfig.sendOnEnter);
+
+    return Shortcuts(
+      shortcuts: useShortCuts
+          ? <LogicalKeySet, Intent>{
+              LogicalKeySet(LogicalKeyboardKey.enter): SubmitIntent(),
```

Comment:
why don't you use `onEditingComplete `or `onSubmitted` of TextField?

Reply:
Because we use TypeAheadField library to show suggestions, it doesn't have the native submit on enter

Reply:
![Screenshot 2023-09-06 at 09 50 23](https://github.com/linagora/twake-on-matrix/assets/12546908/29cb6c38-8c90-4609-8574-f907c9fb3743)
I dont beleive that

Reply:
sorry it was confusing from me, It works but only if you focus the field. I don't know if we want to be able to press enter no matter what element we focus

Reply:
in doubt I used the native "onEditingComplete" to avoid using shortcuts widget which can bug on web version sometimes (cf: https://github.com/linagora/twake-on-matrix/pull/552#issuecomment-1707579400)

Reply:
okay